### PR TITLE
feat(icons): added `list-filter-plus` icon

### PR DESCRIPTION
**Pull request by abdeniz**
## What is the purpose of this pull request?
- [x] New Icon

### Description
Added new `list-filter-plus` icon.

### Icon use case

Use cases: 

- Adding column filters to a data table
- Adding advanced filters

### Concept

- [x] I have provided valid use cases for each icon.
- [x] I have not added any a brand or logo icon.
- [x] I have not used any hate symbols.
- [x] I have not included any religious or political imagery.

### Author, credits & license

- [ ] The icons are solely my own creation.
- [x] The icons were originally created in #<issueNumber> by @abdeniz
- [x] I've based them on the following Lucide icons: `list-filter`, `list-plus`
- [ ] I've based them on the following design: 

### Naming

- [x] I've read and followed the [naming conventions](https://lucide.dev/guide/design/icon-design-guide#naming-conventions)
- [x] I've named icons by what they are rather than their use case.
- [x] I've provided meta JSON files in `icons/[iconName].json`.

### Design

- [x] I've read and followed the [icon design guidelines](https://lucide.dev/guide/design/icon-design-guide)
- [x] I've made sure that the icons look sharp on low DPI displays.
- [x] I've made sure that the icons look consistent with the icon set in size, optical volume and density.
- [x] I've made sure that the icons are visually centered.
- [x] I've correctly optimized all icons to three points of precision.

## Before Submitting
- [x] I've read the [Contribution Guidelines](https://github.com/lucide-icons/lucide/blob/main/CONTRIBUTING.md).
- [x] I've checked if there was an existing PR that solves the same issue.

### CODE REVIEW

File: icons/list-filter-plus.svg [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+  stroke="currentColor"
+  stroke-width="2"
+  stroke-linecap="round"
+  stroke-linejoin="round"
+>
+  <path d="M10 18h4" />
+  <path d="M16 6H3" />
+  <path d="M18 9v6" />
+  <path d="M21 12h-6" />
+  <path d="M7 12h4" />
```
Comment (jguddas) [inline icons/list-filter-plus.svg:16]:
```suggestion
  <path d="M10 18h4" />
  <path d="M15.5 6H3" />
  <path d="M18 9v6" />
  <path d="M21 12h-6" />
  <path d="M7 12h4" />
```

**Comment by jguddas** [comment]
With it in the top corner it's better.
There is a gap violation in this version now, needs to be at least 2px between the plus and top line.

**Comment by karsa-mistmere** [comment]
> With it in the top corner it's better. There is a gap violation in this version now, needs to be at least 2px between the plus and top line.

The plus is also off grid.

**Comment by abdeniz** [comment]
> I'm not really sold on this design, it kinda looks more like `align-right-plus`.

It does indeed, I believe moving the plus to the top line eliminates this confusion, although now it does differ from the `list-plus` icon's plus position. Could something like this work?

image image


image

**Comment by karsa-mistmere** [comment]
> > I'm not really sold on this design, it kinda looks more like `align-right-plus`.
> 
> It does indeed, I believe moving the plus to the top line eliminates this confusion, although now it does differ from the `list-plus` icon's plus position. Could something like this work?
> 
> image image
> 
> image

The design seems fine, but let's address Jakob's concern as well as mine about gaps and keeping lines pixel perfect.

**Comment by abdeniz** [comment]
I believe the middle line still has a gap violation, fixing it

**Comment by abdeniz** [comment]
Fixed 2px gap violation on the middle line:

image

image

**Comment by jguddas** [comment]
> Fixed 2px gap violation on the middle line:
> 
> image
> 
> image
> 

There is a share > copy GitHub embed, you don't have to paste the screenshot.

**Comment by abdeniz** [comment]
I believe I've got the gaps and grid alignment correct now.

<a title="Open lucide studio" target="_blank" href="https://lucide-studio.vercel.app/edit?value=%3Cpath+d=%22M10+18h4%22+/%3E%3Cpath+d=%22M11+6H3%22+/%3E%3Cpath+d=%22M15+6h6%22+/%3E%3Cpath+d=%22M18+9V3%22+/%3E%3Cpath+d=%22M7+12h8.5%22+/%3E">icons<br/>Open lucide studio</a>

**Comment by abdeniz** [comment]
Hi @jguddas @karsa-mistmere, is there anything further I can work on or improve to get this considered?

**Review by ericfennis — APPROVED** [review]

File: icons/list-filter-plus.json [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+    "abdeniz"
+  ],
+  "tags": [
+    "filter",
+    "plus",
+    "options",
+    "add"
+  ],
+  "categories": [
+    "text"
```
Comment (ericfennis) [inline icons/list-filter-plus.json:13]: Can you also add the `layout` category here.

Comment (abdeniz) [reply icons/list-filter-plus.json]: Done ✅

**Review by jguddas — APPROVED** [review]

**Pull request merged**